### PR TITLE
API to refresh access token api for user

### PR DESCRIPTION
**Pull request by SM43**
This adds implementation for API to refresh access token for user. This
requires user refresh token to be passed to get a new access token.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

### CODE REVIEW

File: api/design/auser.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
+// Copyright © 2020 The Tekton Authors.
```
Comment (PuneetPunamiya) [inline api/design/auser.go:1]: Should it be 2021 ??

File: api/design/auser.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+			Attribute("data", accessToken, "User Access JWT")
+			Required("data")
+		})
+
+		HTTP(func() {
+			POST("/user/refresh/accesstoken")
+			Header("refreshToken:Authorization")
+
+			Response(StatusOK)
+			Response("internal-error", StatusInternalServerError)
```
Comment (PuneetPunamiya) [inline api/design/auser.go:52]: Are we missing invalid authorization code error ??

File: api/design/auser.go (hunk 1 above)
Comment (SM43) [reply api/design/auser.go:1]: yep

File: api/design/auser.go (hunk 2 above)
Comment (SM43) [reply api/design/auser.go:52]: why do we need `invalid authorization code error` ? this is not login api

Comment (PuneetPunamiya) [reply api/design/auser.go:52]: Then why are we mentioning it above ??

File: api/pkg/service/user/user.go [hunk 3]
```diff
@@ -0,0 +1,130 @@
+// Copyright © 2020 The Tekton Authors.
```
Comment (PuneetPunamiya) [inline api/pkg/service/user/user.go:1]: Same as above

File: api/design/auser.go (hunk 2 above)
Comment (SM43) [reply api/design/auser.go:52]: ohh.. willl remove it :P

File: api/design/auser.go [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+package design
+
+import (
+	. "goa.design/goa/v3/dsl"
+)
+
+var _ = Service("user", func() {
+	Description("The user service exposes endpoint to get user specific specs")
+
+	Error("invalid-code", ErrorResult, "Invalid Authorization code")
```
Comment (SM43) [inline api/design/auser.go:24]: not needed

File: api/pkg/service/user/user.go [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+	}
+
+	for _, s := range dbUser.Scopes {
+		userScopes = append(userScopes, s.Name)
+	}
+
+	return userScopes, nil
+}
+
+func createChecksum(token string) string {
```
Comment (PuneetPunamiya) [inline api/pkg/service/user/user.go:127]: Can you please add a unit test for this

File: api/design/user.go [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+import (
+	"github.com/tektoncd/hub/api/design/types"
+	. "goa.design/goa/v3/dsl"
+)
+
+var _ = Service("user", func() {
+	Description("The user service exposes endpoint to get user specific specs")
+
+	Error("invalid-token", ErrorResult, "Invalid User token")
+	Error("invalid-scopes", ErrorResult, "Invalid User scope")
```
Comment (PuneetPunamiya) [inline api/design/user.go:26]: I don't think we have used this error in the implementation of the api

File: api/pkg/service/user/user.go (hunk 5 above)
Comment (SM43) [reply api/pkg/service/user/user.go:127]: Do we need a separate test?  this is being tested with the API as we store it in db and check the db record.

File: api/design/user.go (hunk 6 above)
Comment (SM43) [reply api/design/user.go:26]: It is being by the auth service while verifying jwt

File: api/pkg/service/user/user.go (hunk 5 above)
Comment (PuneetPunamiya) [reply api/pkg/service/user/user.go:127]: In db we just check if we have checksum, what I think is passing the token it should return the checksum is what this function does and hence I think we need to test it

Comment (SM43) [reply api/pkg/service/user/user.go:127]: yeah but before checking in db it creates the checksum

**Pull request merged**